### PR TITLE
Cleanup MapReduce package and API

### PR DESCRIPTION
**Pull request by milleruntime**
Please see separate commits for each step so far.  This is still a work in progress but I was looking for feedback before implementing each new MapReduce class.  

The idea for the new API was to have each Accumulo*Format be required to only call one method:
public static void setInfo(Job job, InputInfo info);

The only object required with the Job would be a getter interface object that would have a builder to go along (see InputInfo and InputInfoImpl).  This would only expose these methods and the getter interface object.  Everything else would stay under the hood (and OUT of the public API) allowing us to refactor and clean up the internals as we see fit.

### CODE REVIEW

File: client/mapreduce/src/main/java/org/apache/accumulo/mapreduce/impl/InputInfoImpl.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    SamplerConfiguration samplerConfig;
+    Map<String, String> hints;
+
+    @Override
+    public InputInfo.InputFormatBuilder.TableParams clientInfo(ClientInfo clientInfo) {
+      this.clientInfo = clientInfo;
+      return this;
+    }
+    @Override
+    public InputInfo.InputFormatBuilder.TableParams clientProperties(Properties clientPropsFiles) {
```
Comment (mikewalch) [inline client/mapreduce/src/main/java/org/apache/accumulo/mapreduce/impl/InputInfoImpl.java]: The naming of `clientPropsFiles` should be changed to `clientProps` as there is no file.

However, you could add a `clientPropertiesFile` method and use the code in my PR #715 to get ClientInfo from the file path.

**Comment by milleruntime** [comment]
Sorry for all the rebasing... I did some refactoring and wanted to group that separately from the new stuff.  Should be better now to review.

**Comment by milleruntime** [comment]
I think I am going to shelve these changes for 2.1

**Comment by milleruntime** [comment]
> I think I am going to shelve these changes for 2.1

On second thought, I think I can break up some of the changes into 2 commits.  I will use this PR for the first batch.

**Comment by milleruntime** [comment]
@keith-turner and @ctubbsii please take a look at these 3 commits as the next step in this cleanup process, building on #425 and I can close #720.  These changes have been dialed back to the following:

1. Copy back the MapReduce code from the new client-mapreduce module back into core.client to preserve backwards compatibility of core.
2. Rename the new client-mapreduce packages to avoid conflicting with the old ones
3. Remove the dependency for client-mapreduce from test pom since test is now using the old ones again.  This commit also had sorting and formatting cleanup for the new package names.

This will build on the work already done without breaking things.  Getting these changes will allow someone else to go through deprecating the old MapReduce code while simultaneously working on a new API.

**Comment by ctubbsii** [comment]
Can you rebase, please? There appears to be a conflict with master, and I think a rebase would also clear things up with regard to the diff from master.

**Comment by milleruntime** [comment]
Rebased onto master.

**Comment by ctubbsii** [comment]
So, it looks like adding the classes back to core slows down spotbugs findsecbugs plugin just enough to make that task timeout in Travis due to it taking longer than 10 minutes to analyze core. I've pushed an update to the Travis config to extend the timeout to 20 minutes.

**Pull request merged**